sycl/source/detail/kernel_impl.cpp

```diff
@@ -57,12 +57,32 @@ kernel_impl::~kernel_impl() {
   }
 }
 
+template <info::kernel param>
+typename info::param_traits<info::kernel, param>::return_type
+get_host_kernel_info() {
+  throw runtime_error(
+      "This kernel info parameter is not supported on HOST device.");
+}
+
+template <> string_class get_host_kernel_info<info::kernel::function_name>() {
+  return "Host kernel";
+}
+
+template <> cl_uint get_host_kernel_info<info::kernel::num_args>() { return 0; }
+
+template <> cl_uint get_host_kernel_info<info::kernel::reference_count>() {
+  return 0;
+}
+
+template <> string_class get_host_kernel_info<info::kernel::attributes>() {
```

Comment:
This is not valid output if any attribute is passed with `__attribute__` qualifier to kernel function.

Reply:
These kernel info queues cannot be implemented with pure c++ as returning precise values would require some frontend support. It was suggested to have these values instead of just aborting the process.

Reply:
From my perspective raising exception with "unsupported on host" is more natural for developers and users.